auth/postgres/repo.go

```diff
@@ -127,32 +142,69 @@ func (pr *patRepo) RetrieveAll(ctx context.Context, userID string, pm auth.PATSP
 	return page, nil
 }
 
-func (pr *patRepo) RetrieveSecretAndRevokeStatus(ctx context.Context, userID, patID string) (string, bool, bool, error) {
-	q := `
-		SELECT p.secret, p.revoked, p.expires_at 
+func PageQuery(pm auth.PATSPageMeta) (string, error) {
+	var query []string
+	if pm.Name != "" {
+		query = append(query, "p.name ILIKE '%' || :name || '%'")
+	}
+
+	if pm.ID != "" {
+		query = append(query, "p.id = :id")
+	}
+
+	if pm.Status != auth.AllStatus {
+		switch pm.Status {
+		case auth.RevokedStatus:
+			query = append(query, "p.revoked = TRUE")
+		case auth.ExpiredStatus:
+			query = append(query, "p.revoked = FALSE AND p.expires_at IS NOT NULL AND p.expires_at < :timestamp")
+		case auth.ActiveStatus:
+			query = append(query, "p.revoked = FALSE AND (p.expires_at IS NULL OR p.expires_at >= :timestamp)")
+		}
+	}
+
+	var emq string
+	if len(query) > 0 {
+		emq = fmt.Sprintf("AND %s", strings.Join(query, " AND "))
+	}
+	return emq, nil
+}
+
+func (pr *patRepo) RetrieveSecretAndRevokeStatus(ctx context.Context, userID, patID string) (string, auth.Status, error) {
```

Comment:
Why `RetrieveSecretAndRevokeStatus` what `auth.Status`, instead it can return TRUE or FALSE as boolean